tests/test_httpstatus.py

```diff
@@ -0,0 +1,196 @@
+
+    def test_raise_status_in_before_hook(self):
+        """ Make sure we get the 200 raised by before hook """
+        body = self.simulate_request('/status', method='GET', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_in_responder(self):
+        """ Make sure we get the 200 raised by responder """
+        body = self.simulate_request('/status', method='POST', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_runs_after_hooks(self):
+        """ Make sure after hooks still run """
+        body = self.simulate_request('/status', method='PUT', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_survives_after_hooks(self):
+        """ Make sure after hook doesn't overwrite our status """
+        body = self.simulate_request('/status', method='DELETE',
+                                     decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_empty_body(self):
+        """ Make sure passing None to body results in empty body """
+        body = self.simulate_request('/status', method='PATCH', decode='utf-8')
+        self.assertEqual(body, '')
+
+
+class TestHTTPStatusWithGlobalHooks(testing.TestBase):
+    def before(self):
+        self.resource = TestHookResource()
+
+    def test_raise_status_in_before_hook(self):
+        """ Make sure we get the 200 raised by before hook """
+        self.api = falcon.API(before=[before_hook])
+        self.api.add_route('/status', self.resource)
+
+        body = self.simulate_request('/status', method='GET', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_runs_after_hooks(self):
+        """ Make sure we still run after hooks """
+        self.api = falcon.API(after=[after_hook])
+        self.api.add_route('/status', self.resource)
+
+        body = self.simulate_request('/status', method='GET', decode='utf-8')
+        self.assertEqual(self.srmock.status, falcon.HTTP_200)
+        self.assertIn(('x-failed', 'False'), self.srmock.headers)
+        self.assertEqual(body, 'Pass')
+
+    def test_raise_status_survives_after_hooks(self):
+        """ Make sure after hook doesn't overwrite our status """
+        self.api = falcon.API(after=[noop_after_hook])
```

Comment:
The global after hook would be expected to overwrite the status. Therefore, it seems like this test should use an after hook that attempts to do so, and ensures it succeeds, as in `test_raise_status_runs_process_response`.